src/StreamMap.php

```diff
@@ -13,13 +13,13 @@ class StreamMap
 	 * @param VideoInfo $video_info
 	 * @return StreamMap
 	 */
-	public static function createFromVideoInfo(VideoInfo $video_info)
+	public static function createFromVideoInfo(VideoInfo $video_info, $video_id)
 	{
 		// get the url_encoded_fmt_stream_map, and explode on comma
 		$streams = explode(',', $video_info->getStreamMapString());
 		$formats = explode(',', $video_info->getAdaptiveFormatsString());
 
-		return new self($streams, $formats);
+		return new self($streams, $formats, $video_id);
```

Comment:
You can use $video_info->getVideoId() here:
`return new self($streams, $formats, $video_info->getVideoId());`